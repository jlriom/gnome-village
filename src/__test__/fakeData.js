export const category = {
    category: "category",
    count: 0
}

export const habitantsByHairColor = [
    {
        hair_color: 'Black',
        count: 256
    },
    {
        hair_color: 'Gray',
        count: 258
    },
    {
        hair_color: 'Green',
        count: 267
    },
    {
        hair_color: 'Pink',
        count: 285
    },
    {
        hair_color: 'Red',
        count: 273
    }
];

export const habitantsByProfession = [
    {
        profession: 'Baker',
        count: 214
    },
    {
        profession: 'Blacksmith',
        count: 180
    },
    {
        profession: 'Brewer',
        count: 200
    },
    {
        profession: 'Butcher',
        count: 210
    },
    {
        profession: 'Carpenter',
        count: 216
    },
    {
        profession: 'Cook',
        count: 190
    },
    {
        profession: 'Farmer',
        count: 205
    },
    {
        profession: 'Gemcutter',
        count: 186
    },
    {
        profession: 'Leatherworker',
        count: 185
    },
    {
        profession: 'Marble Carver',
        count: 190
    },
    {
        profession: 'Mason',
        count: 214
    },
    {
        profession: 'Mechanic',
        count: 182
    },
    {
        profession: 'Medic',
        count: 187
    },
    {
        profession: 'Metalworker',
        count: 192
    },
    {
        profession: 'Miner',
        count: 201
    },
    {
        profession: 'Potter',
        count: 202
    },
    {
        profession: 'Prospector',
        count: 202
    },
    {
        profession: 'Sculptor',
        count: 182
    },
    {
        profession: 'Smelter',
        count: 203
    },
    {
        profession: 'Stonecarver',
        count: 187
    },
    {
        profession: 'Tailor',
        count: 199
    },
    {
        profession: 'Tax inspector',
        count: 192
    },
    {
        profession: ' Tinker',
        count: 202
    },
    {
        profession: 'Woodcarver',
        count: 199
    }
];

export const auth = { name: "name", isGuest: false };

export const habitant = {
    id: 362,
    name: 'Cogwitz Bittergrill',
    thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
    age: 127,
    weight: 40.074535,
    height: 93.66314,
    hair_color: 'Black',
    professions: [
        'Prospector',
        ' Tinker',
        'Baker',
        'Tailor',
        'Leatherworker',
        'Sculptor'
    ],
    friends: [
        'Whitwright Fussbuster',
        'Fizwood Bitterrocket',
        'Fizwood Felpower'
    ]
};

export const friends = [
    {
        id: 363,
        name: 'Cogwitzs friend ',
        thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
        age: 127,
        weight: 40.074535,
        height: 93.66314,
        hair_color: 'Black',
        professions: [
            'Prospector',
            ' Tinker'
        ],
        friends: [
            'Whitwright Fussbuster'
        ]
    }
];
