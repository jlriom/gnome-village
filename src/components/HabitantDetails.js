import React from "react";
import PropTypes from "prop-types";
import PageLayout from "./layout/PageLayout";
import HabitantItem from "./HabitantItem";
import HabitantCard from "./HabitantCard";

const HabitantDetails = ({ profile, friends }) => {
	return (
		<PageLayout headerText={profile && profile.name}>
			<div className='container'>
				<div className='row'>
					<div className='col-3'>
						<HabitantCard habitant={profile} />
					</div>
					<div className='col-9'>
						<h3 className='m-4'>Friends</h3>
						{friends &&
							friends.map(friend => {
								return (
									<HabitantItem
										key={friend.id}
										habitant={friend}
										manageFriends={false}
									/>
								);
							})}
						<br></br>
					</div>
				</div>
			</div>
		</PageLayout>
	);
};

HabitantDetails.propTypes = {
	profile: PropTypes.object.isRequired,
	friends: PropTypes.array.isRequired
};

export default HabitantDetails;
