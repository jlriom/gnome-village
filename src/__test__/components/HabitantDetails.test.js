import React from "react";
import HabitantDetails from "../../components/HabitantDetails";
import renderer from "react-test-renderer";
import { habitant, friends } from "../fakeData";
import { MemoryRouter } from 'react-router-dom';

describe("<HabitantDetails />", () => {
    it("renders ok", () => {
        const tree = renderer.create(
            <MemoryRouter>
                <HabitantDetails profile={habitant} friends={friends} />
            </MemoryRouter>
        );
        expect(tree).toMatchSnapshot();
    });
});
