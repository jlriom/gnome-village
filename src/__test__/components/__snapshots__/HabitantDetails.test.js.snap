// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HabitantDetails /> renders ok 1`] = `
Array [
  <h2
    className="my-3 display-4"
  >
    Cogwitz Bittergrill
  </h2>,
  <hr />,
  <div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-3"
        >
          <div
            className="card"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <div
              className="lazyload-placeholder"
              style={
                Object {
                  "height": undefined,
                }
              }
            />
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
              >
                Cogwitz Bittergrill
              </div>
              <div
                className="container"
                style={
                  Object {
                    "weight": "100%",
                  }
                }
              >
                <dl
                  className="row"
                >
                  <dt
                    className="col-sm-6"
                  >
                     
                    Age:
                  </dt>
                  <dd
                    className="col-sm-6"
                  >
                     
                    127
                  </dd>
                  <dt
                    className="col-sm-6"
                  >
                     
                    Hair Color:
                  </dt>
                  <dd
                    className="col-sm-6"
                  >
                     
                    Black
                  </dd>
                  <dt
                    className="col-sm-6"
                  >
                     
                    Weight:
                  </dt>
                  <dd
                    className="col-sm-6"
                  >
                     
                    40.075
                  </dd>
                  <dt
                    className="col-sm-6"
                  >
                     
                    Height:
                  </dt>
                  <dd
                    className="col-sm-6"
                  >
                     
                    93.663
                  </dd>
                </dl>
                <hr />
                <div
                  className="row"
                >
                  <dl
                    className="row"
                  >
                    <dt
                      className="col-sm-6"
                    >
                      Professions:
                    </dt>
                    <dd
                      className="col-sm-6"
                    >
                        
                    </dd>
                    <dt
                      className="col-sm-6"
                    >
                       
                    </dt>
                    <dd
                      className="col-sm-6 "
                    >
                      Prospector
                    </dd>
                    <dt
                      className="col-sm-6"
                    >
                       
                    </dt>
                    <dd
                      className="col-sm-6 "
                    >
                       Tinker
                    </dd>
                    <dt
                      className="col-sm-6"
                    >
                       
                    </dt>
                    <dd
                      className="col-sm-6 "
                    >
                      Baker
                    </dd>
                    <dt
                      className="col-sm-6"
                    >
                       
                    </dt>
                    <dd
                      className="col-sm-6 "
                    >
                      Tailor
                    </dd>
                    <dt
                      className="col-sm-6"
                    >
                       
                    </dt>
                    <dd
                      className="col-sm-6 "
                    >
                      Leatherworker
                    </dd>
                    <dt
                      className="col-sm-6"
                    >
                       
                    </dt>
                    <dd
                      className="col-sm-6 "
                    >
                      Sculptor
                    </dd>
                  </dl>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-9"
        >
          <h3
            className="m-4"
          >
            Friends
          </h3>
          <span
            className="d-inline-block m-4 full-width"
          >
            <div
              className="full-width media"
            >
              <div
                className="lazyload-placeholder"
                style={
                  Object {
                    "height": undefined,
                  }
                }
              />
              <div
                className="full-width media-body"
              >
                <h5
                  className="d-flex justify-content-between"
                >
                  <span>
                    Cogwitzs friend 
                  </span>
                   
                </h5>
                <hr />
                <div
                  className="container"
                >
                  <small>
                    <div
                      className="row"
                    >
                      <div
                        className="col-lg-4"
                      >
                        <dl
                          className="row full-width"
                        >
                          <dt
                            className="col-sm-6"
                          >
                             
                            Age:
                          </dt>
                          <dd
                            className="col-sm-6"
                          >
                             
                            127
                          </dd>
                          <dt
                            className="col-sm-6"
                          >
                             
                            Hair Color:
                          </dt>
                          <dd
                            className="col-sm-6"
                          >
                             
                            Black
                          </dd>
                          <dt
                            className="col-sm-6"
                          >
                             
                            Weight:
                          </dt>
                          <dd
                            className="col-sm-6"
                          >
                             
                            40.075
                          </dd>
                          <dt
                            className="col-sm-6"
                          >
                             
                            Height:
                          </dt>
                          <dd
                            className="col-sm-6"
                          >
                             
                            93.663
                          </dd>
                        </dl>
                      </div>
                      <div
                        className="col-lg-4"
                      >
                        <div
                          className="row full-width"
                        >
                          <b>
                            Professions: 
                          </b>
                           
                          <span>
                            Prospector
                            , 
                          </span>
                          <span>
                             Tinker
                            
                          </span>
                        </div>
                      </div>
                      <div
                        className="col-lg-4"
                      >
                        <div
                          className="row full-width"
                        >
                          <b>
                            Friends: 
                          </b>
                            
                          <span>
                            <a
                              href="/brastlewark/Whitwright Fussbuster"
                              onClick={[Function]}
                            >
                              Whitwright Fussbuster
                            </a>
                            
                          </span>
                        </div>
                      </div>
                    </div>
                  </small>
                </div>
              </div>
            </div>
          </span>
          <br />
        </div>
      </div>
    </div>
  </div>,
]
`;
